### PR TITLE
bump h2, postgresql and hikariCP versions

### PR DESCRIPTION
**Pull request by KadekM**
Since doobie 0.3 depends on jdk8+ , can't we use hikariCP for java8 for 0.3.1 ?

### CODE REVIEW

**Comment by tpolecat** [comment]
Ah good catch. Thanks!

**Comment by tpolecat** [comment]
It looks like the Postgres driver may have added a new JDBC type. Odd that it only turned up in the 2.11 build. I'm going to flush the caches and rebuild.

**Comment by KadekM** [comment]
Yea the fail only on scala 2_11 weirded me out, my first thought was that
we run Analysis tests only on 2_11 but it's hard to check from cellphone :)

**Comment by tpolecat** [comment]
Yeah it looks like there's a driver change. We'll need to investigate.

**Comment by KadekM** [comment]
I've gathered all changes between driver versions (1201, 1211] in chronological order (on top of file changes from 1202, then 1203 etc...):
https://gist.github.com/KadekM/4318774344d90f68f132fb62a816d642

First file is narrowed down to what I guess may interest us, second file is all PRs unfiltered (including style, tests etc...)

I thought it may help us.

**Comment by KadekM** [comment]
It seems to have been refcursor

File: yax/postgres/src/main/scala/doobie/postgres/free/pgconnection.scala [hunk 1]
```diff
@@ -83,7 +85,7 @@ object pgconnection {
 
     // Primitive Operations
     case class  AddDataType(a: String, b: String) extends PGConnectionOp[Unit]
-    case class  AddDataType1(a: String, b: Class[_]) extends PGConnectionOp[Unit]
+    case class  AddDataType1(a: String, b: Class[_ <: PGobject]) extends PGConnectionOp[Unit]
```
Comment (tpolecat) [inline yax/postgres/src/main/scala/doobie/postgres/free/pgconnection.scala]: Why is this upper bound needed? Did the driver change the `.addDataType` method?

**Comment by tpolecat** [comment]
Do you know which test was failing due to the missing `RefCursor` constant? I don't recall writing a test that uses it.

**Comment by KadekM** [comment]
Everything in AnalysisTestSpec...

```
  check(AnalysisTest.speakerQuery2)
  check(AnalysisTest.arrayTest)
  check(AnalysisTest.pointTest)
  check(AnalysisTest.update2)
```
so even the simplest queries that have nothing to do with it... i.e. this would fail (but only the check of AnalysisSpec)

```
sql"""SELECT C.NAME FROM COUNTRYLANGUAGE""".query[String]
```

**Comment by tpolecat** [comment]
Ah I gotcha, the problem is when it's loading up driver metadata. The database is saying it supports a mapping for `RefCursor` and we didn't recognize it. Makes sense now.

**Comment by KadekM** [comment]
I see. Thanks for info :)

**Comment by tpolecat** [comment]
Anyway thanks for this! 👍

File: yax/postgres/src/main/scala/doobie/postgres/free/pgconnection.scala (hunk 1 above)
Comment (KadekM) [inline yax/postgres/src/main/scala/doobie/postgres/free/pgconnection.scala]: Yes they did.
`void addDataType(String type, Class<? extends PGobject> klass) throws SQLException;`

Comment (KadekM) [inline yax/postgres/src/main/scala/doobie/postgres/free/pgconnection.scala]: Before it was just `Class<?>`

**Pull request merged**